python/lsst/daf/butler/core/datasets/type.py

```diff
@@ -222,6 +222,15 @@ def __hash__(self) -> int:
         return hash((self._name, self._dimensions, self._storageClassName,
                      self._parentStorageClassName))
 
+    def __lt__(self, other: Any) -> bool:
+        """Sort using the dataset type name.
+        """
+        if not isinstance(other, type(self)):
+            other_name = str(other)
```

Comment:
What kind of heterogenous sorting is this intended to support?  I would have thought this could safely raise `TypeError` or something.

Reply:
I can fail here no problem. I was mostly trying to not blow up if somebody ended up with a DatasetType and DatasetRef somehow in the same collection.